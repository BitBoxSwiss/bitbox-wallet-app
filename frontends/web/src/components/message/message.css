.message {
    border: solid 1px var(--color-brightgray);
    color: var(--color-secondary);
    border-radius: 2px;
    padding: var(--spacing-default);
    margin-bottom: var(--spacing-default);
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: flex-start;
    flex-shrink: 0;
    overflow: hidden;
}

.message img {
    margin-right: var(--spacing-default);
}

.message p:last-child {
    margin-bottom: 0 !important;
}

.success {
    composes: message;
    color: var(--color-success);
    border-color: var(--color-success);
}

.info {
    composes: message;
    color: var(--color-default);
    border-color: var(--color-default);
}

.warning {
    composes: message;
    color: var(--color-warning);
    border-color: var(--color-warning);
}

.error {
    composes: message;
    color: var(--color-error);
    border-color: var(--color-error);
}
