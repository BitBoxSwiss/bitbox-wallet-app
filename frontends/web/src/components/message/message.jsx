/**
 * Copyright 2018 Shift Devices AG
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { h } from 'preact';
import styles from './message.css';

export default function Message({
    type,
    style,
    children,
}) {
    return (
        <div className={styles[type] || styles.message} style={style}>
            {children}
        </div>
    );
}
