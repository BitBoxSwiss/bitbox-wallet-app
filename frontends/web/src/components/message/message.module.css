.message {
    align-items: flex-start;
    border: solid 1px var(--background-quaternary);
    border-radius: 2px;
    color: var(--color-default);
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    justify-content: flex-start;
    margin-bottom: var(--spacing-default);
    overflow: hidden;
    padding: var(--spacing-default);
}

.content {
    padding-top: 2px;
    width: 100%;
}

.message .title {
    margin: 0 0 var(--spacing-half) 0;
}

@media (max-width: 768px) {
    .message .title {
        --size-subheader: 18px;
    }
}

.message.small {
    flex-shrink: 1;
    font-size: var(--size-default);
}

.message img {
    margin-right: 4px;
    max-width: 30px;
}

.message p:last-child {
    margin-bottom: 0 !important;
}

.success {
    composes: message;
    background-color: var(--color-success-background);
    border-color: var(--color-success-border);
}

.info {
    composes: message;
    background-color: var(--color-info-background);
    border-color: var(--color-info-border);
}

.warning {
    composes: message;
    background-color: var(--color-warning-background);
    border-color: var(--color-warning-border);
}

.error {
    composes: message;
    background-color: var(--color-error-background);
    border-color: var(--color-error-border);
}