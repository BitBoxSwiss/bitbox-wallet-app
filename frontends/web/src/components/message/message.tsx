/**
 * Copyright 2018 Shift Devices AG
 * Copyright 2024 Shift Crypto AG
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { ReactNode } from 'react';
import { StatusInfo, StatusSuccess, StatusWarning, StatusError } from '@/components/icon';
import { TMessageTypes } from '@/utils/types';
import styles from './message.module.css';

type TMessageIconProps = { type: TMessageTypes };

const MessageIcon = ({ type }: TMessageIconProps) => {
  switch (type) {
  case 'success':
    return (
      <StatusSuccess />
    );
  case 'info':
    return (
      <StatusInfo />
    );
  case 'error':
    return (
      <StatusError />
    );
  case 'warning':
    return (
      <StatusWarning />
    );
  default:
    return null;
  }
};

type MessageProps = {
  hidden?: boolean;
  small?: boolean;
  title?: string;
  type?: TMessageTypes;
  noIcon?: boolean;
  children: ReactNode;
}

export const Message = ({
  hidden,
  small,
  title,
  type = 'info',
  noIcon = false,
  children,
}: MessageProps) => {
  if (hidden) {
    return null;
  }
  return (
    <div className={`${styles[type]} ${small ? styles.small : ''}`}>
      {!noIcon && <MessageIcon type={type} />}
      <div className={styles.content}>
        {title && (
          <h2 className={`subTitle ${styles.title}`}>{title}</h2>
        )}
        {children}
      </div>
    </div>
  );
};
