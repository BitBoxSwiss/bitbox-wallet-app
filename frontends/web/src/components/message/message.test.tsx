/**
 * Copyright 2018 Shift Devices AG
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { Message } from './message';
import '@testing-library/jest-dom'; // TODO needed?

describe('components/message/message', () => {
  it('should have child nodes', () => {
    const msg = render(<Message><span>hello</span></Message>);
    expect(msg.container.firstChild!).toMatchSnapshot();
  });

  it('should return return null', () => {
    const msg = render(<Message hidden><span>hello</span></Message>);
    expect(msg.container).toBeEmptyDOMElement();
  });

  it('should preserve text', () => {
    const msg = render(<Message><span>hello world</span></Message>);
    expect(msg.findByText('hello world')).not.toBeNull();
  });
});
