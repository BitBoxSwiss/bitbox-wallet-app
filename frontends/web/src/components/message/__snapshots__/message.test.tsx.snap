// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/message/message > should have child nodes 1`] = `
<div
  class="_message_88ce03 "
>
  <span>
    hello
  </span>
</div>
`;
